include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -15,11 +15,21 @@
 
 #include <optional>
 
+#include <seqan3/core/algorithm/configuration.hpp>
 #include <seqan3/core/type_traits/template_inspection.hpp>
 
 namespace seqan3::detail
 {
 
+// forward declaration for friend declaraion in alignment_result.
+template <typename>
+#if !SEQAN3_WORKAROUND_GCC_93467
+//!\cond
+    requires is_type_specialisation_of_v<configuration_t, configuration>
+//!\endcond
+#endif // !SEQAN3_WORKAROUND_GCC_93467
+class policy_alignment_result_builder;
```

Comment:
EDIT: I think I misread your intention. You want to add this constraint, but gcc doesn't let you, right?

---

I'm not sure if this really a bug, because forward declaring something should have the same template constraints, but let's see what the consensus of clang, gcc ans msvc is:

https://godbolt.org/z/H4FfSC

I would expect that a forward declare must have the same constraints, otherwise you could get inconsistencies when using that forward declared struct.

The gcc [ticket 93467](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93467) quotes the standard:

> "A constrained declaration may only be redeclared using the same syntactic form. No diagnostic is required."

Reply:
Ah, the edit came after your comment. Now it makes sense. Right, gcc doesn't let me although the standard requires it.